std/math/index.zig

```diff
@@ -242,6 +242,82 @@ test "math.max" {
     assert(max(i32(-1), i32(2)) == 2);
 }
 
+/// This function returns the minimum value of the integer type T.
+/// The result is a compile time constant.
+/// Tested in test/cases/misc.zig
+pub fn minValue(comptime T: type) @typeOf(42) {
+    comptime {
+        return switch (T) {
+            u1, u2, u3, u4, u5, u6, u7,
+            u8, u16, u32, u64, u128 => 0,
+            i1 => -1,
+            i2 => -2,
+            i3 => -4,
+            i4 => -8,
+            i5 => -16,
+            i6 => -32,
+            i7 => -64,
+            i8 => -128,
+            i16 => -32768,
+            i32 => -2147483648,
+            i64 => -9223372036854775808,
+            i128 => -170141183460469231731687303715884105728,
+            else => {
+                //Calculate for Integers that we do not have cached
```

Comment:
I don't think there is any reason for this explicit caching since the compiler already does this.

Reply:
What does "the compiler already does this" mean here? we are in `comptime`, so obviously we are not caching at runtime, but I thought that event at `comptime`, having a `switch` for common types could speed-up compile times.

Reply:
Well, each instantiation of `min/maxInt` only gets its body comptime evaluated once and then cached. This is how types such as `ArrayList(u8)` is the same type each time it is called. Isn't the point of `comptime` stuff that we can avoid hardcoding our comptime generated tables and such? :)

Reply:
@Hejsil I didn't realize that all return values of `comptime` were cached. Is that defined somewhere in the documentation? I was aware of using `comptime` to create lookup tables, etc for `crc32` [[0](https://github.com/ziglang/zig/blob/master/std/hash/crc.zig#L24)].

I did a test calling `std.math.maxInt(u32)` repeatedly and it did cache, so I will remove the `switch`.